### PR TITLE
Casing conversions

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**

### CODE REVIEW

File: Source/Languages/Casing/CamelCaseConverter.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        /**
+         * Converts a name to camelCase.
+         * 
+         * @param name   A name to convert.
+         * @returns The name's equivalent in camelCase.
+         */
+        public convert(name: string): string {
+            name = super.convert(name);
+
+            return name[0].toLowerCase() + name.substring(1);
```
Comment (faulda) [inline Source/Languages/Casing/CamelCaseConverter.ts]: So this `convert` just calls super.convert, then lowercases the first letter in returns. Does that mean super.convert will return PascalCase, lower cases the first letter, gets camelCase?
If so, does that mean CaseStyleConverter always does PascalCase by default?

File: Source/Languages/Casing/CaseStyleConverter.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+         * @returns The name's equivalent in this converter's case style.
+         */
+        public convert(name: string): string {
+            let position: number = 0;
+            let nameConverted: string = "";
+
+            while (position < name.length) {
+                let nextWordStart = this.findNextWordStart(name, position);
+                let nextWord = name.substring(position, nextWordStart);
+                nextWord = nextWord.replace(/_/g, "").toLowerCase();
```
Comment (faulda) [inline Source/Languages/Casing/CaseStyleConverter.ts]: what does .repace(/_/g, "") do?

File: Source/Languages/Casing/CamelCaseConverter.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [inline Source/Languages/Casing/CamelCaseConverter.ts]: Precisely.

The only difference between camelCase and PascalCase is the first letter of the first word.

File: Source/Languages/Casing/CaseStyleConverter.ts (hunk 2 above)
Comment (JoshuaKGoldberg) [inline Source/Languages/Casing/CaseStyleConverter.ts]: Removes all instances of `"_"`.

JavaScript's string::replace only removes the first instance when given a string, so you have to give it a regular expression to catch them all. Dumb language quirk.

File: Source/ConversionContext.ts [hunk 3]
```diff
@@ -106,6 +107,13 @@ namespace GLS {
         }
 
         /**
+         *
```
Comment (JoshuaKGoldberg) [inline Source/ConversionContext.ts]: Missed the docs here...

**Comment by faulda** [comment]
No further comments :)

**Pull request merged**